### PR TITLE
Advanced Building Blocks - Enumerables

### PR DESCRIPTION
**Pull request by dansantander**
This project recreates the following methods included in the Ruby's Enumerable module:

+each
+each with index
+select
+all?
+any?
+none?
+count
+map
+inject

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**HI @dansantander, you have done a good job:+1:. Now take care of the following :fast_forward:**

1. **my_select** is returning an array of arrays, making the output different from ruby **select** method.
`[1, 2, 3, 4, 5].my_select {|x| x > 2}`  is returning `[[3], [4], [5]]` instead of `[3, 4, 5]`.

2. **my_all?**: when no block or argument is provided, it should return true if none of the collection is false or nil.
`[1, 2, 3].my_all?` is returning `false` instead of `true`.

3. **my_any?:** when no block or argument is given, it should return true if at least one of the elements is true (that is, not false or nil).
`[1, 2, nil, false].my_any?` is returning `false` instead of `true`

4. **my_none?:** when no block or argument is given, it should return true **only if none of** the collection is true.
`[1, 2, 3, false].my_none?` is returning `true` instead of `false`

_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

File: methods.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each do |i|
+      arr << [i] if yield(i)
+    end
+    arr
+  end
```
Comment (euqueme) [inline methods.rb]: #### my_select 

- [ ] returns an array containing all elements of enum for which the given block returns a true value

Given

  block = proc { |num| num < (0 + 9) / 2 } 

Expected
   
>  [2,4,7,11].my_select(&block) should be equal to [2,4,7,11].select(&block)   => [2] not [[2]]

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @dansantander :handshake: 

### Thank you for doing the changes requested by the previous reviewer, :medal_sports: 

### you still have to work on your my_select, :construction: :construction_worker_man:  the results are good but you are returning an array for each element of the enum when you only have to return an array with the elements that pass the condition given in the block. I left a comment in your code :point_down: 

[OPTIONAL]

I think if we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
To make this work we have to use block.call(element) instead of yield(element) in my_any?

### That's it! :raised_hands: 

### Please :pray: do the changes :man_technologist:  and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @dansantander, :tada: 

your project is good to pass you might merge this branch :clap: :heavy_check_mark: 

PROJECT STATUS: APPROVED :green_apple:

**Pull request merged**